xdsl/utils/diagnostic.py

```diff
@@ -16,14 +16,9 @@ def add_message(self, op: Operation, message: str) -> None:
         """Add a message to an operation."""
         self.op_messages.setdefault(op, []).append(message)
 
-    def raise_exception(
-        self,
-        message: str,
-        ir: IRNode,
-        exception_type: type[Exception] = DiagnosticException,
-        underlying_error: Exception | None = None,
-    ) -> NoReturn:
-        """Raise an exception, that will also print all messages in the IR."""
+    def get_output(self, ir: IRNode) -> str:
+        """Get all messages in the IR."""
+
```

Comment:
This is lready a separate commit if someone wants to split it.

It is a basic factoring of the string building method, unmodified, to use it without necessarily raising an exception containing it.

Reply:
What's the difference with `f"{ir}"` in the case of Operation? 

``` python
    def __str__(self) -> str:
        from xdsl.printer import Printer

        res = StringIO()
        printer = Printer(stream=res)
        printer.print_op(self)
        return res.getvalue()
```

Seems like it's only used for Operation below, anyway?

Reply:
Main difference is the diagnostic parameter to the printer:
```
p = Printer(stream=f, diagnostic=self, print_generic_format=True)
```
Printing the IR with diagnostic messages inline.

Reply:
(Also the generic syntax - because it's not generally safe to print custom syntax of unverifying IR, this can mess up error reporting quite a lot!)

Reply:
Ah sounds good